### DISCUSSION TITLE
Is it possible to create multiple rooms through the use of entry keys?

### DISCUSSION THREAD

**Discussion by b3aribeiro** [jbakse/p5.party#48] [Q&A] [answered]
Can I request a key to each participant (for instance, EX3UJ) when entering so they can be directed to a specific room? 
If not, could we request this feature?

**Comment by jbakse** [comment, marked as answer]
We are considering how p5.party might share a unique id for each connected client/participant. Its not currently provided but may be soon. That said, I think the core of your question is "how can I assign players to specific rooms". 

Can you elaborate on that a bit. What, specifically, do you have in mind? Walk me through the user experience, etc. Having a specific use example will help in designing features to support it, or maybe this is something that can be done already and should be shown in an example.

**Reply by b3aribeiro** [reply]
I'm imagining something like this:

1. I access the game page.
2. I see a pop-up or input field where I need to write/create a room key. 
3. If I type an existing room key, it takes me to that room and the game starts.
4. If someone else types a different key that doesn't exist yet they will either be 
   (a) creating a private room or 
   (b) receiving feedback: "invalid key"

**Reply by jbakse** [reply]
I've posted an example of one way to approach this. 

https://p5party.netlify.app/show_example.html?select_room

Please take a look at it, test it out to see if it works as expected, and help to improve the clarity of the example code and README.

**Reply by b3aribeiro** [reply]
This is exactly what I was looking for and it worked!! 

The description and code are very clear. I don't think we need to add anything else to those. 

However, I'm wondering if we can keep the typed room key available for the players once they are in. 
What I noticed is that, once you type a key and click "Join Room", it changes the input field back to main. 
This might be a bit counter-intuitive and participants will surely forget their keys. 
Do you think it's worth adding that?

**Reply by jbakse** [reply]
Yes, that feature is worth adding. That sounds like a good feature request. Want to submit it as an issue?

**Reply by jbakse** [reply]
Moved to issue #52
